node/src/tasks/bsp_upload_file.rs

```diff
@@ -766,7 +787,28 @@ where
                     .extrinsic_retry_timeout,
             ))
             .watch_for_success(&self.storage_hub_handler.blockchain)
-            .await?;
+            .await;
+
+        if let Err(e) = result {
+            debug!(
+                target: LOG_TARGET,
+                "Failed to volunteer for file {:?}: {:?}",
+                file_key,
+                e
+            );
+
+            // Delete file metadata from file storage.
+            let mut write_file_storage = self.storage_hub_handler.file_storage.write().await;
+            write_file_storage
+                .delete_file(&file_key.into())
+                .map_err(|e| anyhow!("Failed to delete file from file storage: {:?}", e))?;
+
+            // Unregister the file from the file transfer service.
+            self.storage_hub_handler
+                .file_transfer
+                .unregister_file(file_key.into())
+                .await?;
+        }
```

Comment:
This fixes a bug where if the another storage request were to be issued after failing the first volunteer check, the second volunteer would fail due to the file already being in the file storage with the error FileAlreadyExists.